src/outputs/outputs.hpp

```diff
@@ -124,6 +125,23 @@ class OutputType {
   OutputData *plast_data_;  // ptr to tail OutputData node in doubly linked list
 };
 
+//----------------------------------------------------------------------------------------
+// Helper definitions to enroll user output variables
+
+// Function signature for currently supported user output functions
+using HstFun_t = std::function<Real(MeshData<Real> *md)>;
+
+// Container
+struct HistoryOutputVar {
```

Comment:
I would add a constructor to this struct. e.g.,
```C++
HistoryOutputVar(const UserHistoryOperation &hst_op_, const HstFun_t &hst_fun_, const std::string &label_)
  : hst_op(hst_op_), hst_fun(hst_fun_), label(label_) { }
```
just to minimize unnecessary copies and remove the need for the list initialization